diagrams/sequence-diagrams/creating-a-form.md

```diff
@@ -0,0 +1,216 @@
+## Creating a form
+
+```mermaid
+sequenceDiagram
+  title Creating a new form
+  actor user as form creator
+  participant browser as Web Browser
+  participant admin as forms-admin
+  link admin: GitHub repo @ https://github.com/alphagov/forms-admin
+  participant api as forms-api
+  link api: GitHub repo @ https://github.com/alphagov/forms-api
+
+  note right of user: user has already logged in
+
+  user->>browser: click "Create a form >" button
+  browser->>admin: GET /forms/new
+  browser-->>user: show "What is the name of your form?" page
+  user->>browser: Give the form a name<br />click "Save and continue" button
+  browser->>admin: POST /forms/new<br />payload: forms_change_name_form%5Bname%5D={form name}
+  admin->>api: POST /forms<br />{"name":"{form name}",<br />"submission_email":"",<br />"org":"{user orginisation}"}
+  api->>api: Create form
+  api-->>admin: {"id":{form id}}
+  admin-->>browser: 302
+  browser->>admin: GET /forms/{form id}
+  Note right of admin: unclear why this next request happens twice, but it does
+  admin->>api: GET /forms/{form id}
+  admin->>api: GET /forms/{form id}
+  admin->>api: GET /forms/{form id}/pages
```

Comment:
For context, the show "Create a form" page has a service that needs to know whether a form has any pages to work out what status to display next the Add/edit question task https://github.com/alphagov/forms-admin/blob/main/app/service/task_status_service.rb#L16-L18